controllers/inventory-controller.js

```diff
@@ -69,7 +68,77 @@ const getItemById = async (req, res) => {
   }
 };
 
+//POST a new inventory item
+const postNewInventoryItem = async (req, res) => {
+  try {
+    const {
+      item_name,
+      description,
+      category,
+      status,
+      quantity,
+      warehouse_name,
+    } = req.body;
+
+    // Validate data
+    if (
+      !item_name ||
+      !description ||
+      !category ||
+      !status ||
+      quantity === undefined ||
+      !warehouse_name
+    ) {
+      return res.status(400).json({ message: "Missing required fields" });
+    }
+    // Check if quantity is a valid number
+    if (isNaN(parseInt(quantity, 10))) {
```

Comment:
Line 95: we don't need to use parseInt inside isNaN. isNaN by itself will automatically attempt to convert whatever being checked into a number (e.g. '1' => 1 or 'something' => ??). Although there are some edge cases that isNaN would fail to catch but I think is good enough for this assignment. So in short. it is enough to just write:

`if (isNaN(quantity)`

Reply:
I added parseInt to account for cases if user adds something like "123abc", or a decimal number like "37.72". Having just NaN will not catch those cases, and parseInt is added here to ensure that only clean data gets written into our database.

Reply:
for '123abc' cases, parseInt won't catch it. It actually stops parsing once it reach 'a'. So we'll end up getting number 123 and accept the good part rather than rejecting the input. ('1a2b3c' will come out as 1 through parseInt)

It's true that isNaN won't catch double. If you want to have tighter validation here, we can consider using regex (regular expression), for example, below regex will match only whole number, not double or mixture of numbers and letters

`/^-?\d+(\d+)?$/`

Reply:
okay, yeah I see your point. I was thinking of cases when a letter or character is accidenatlly added after the quantity, but yes ParseInt will not reject the input, and will just take the good part, which doen't seem ideal. I'll try the regex you suggested.

Reply:
Thanks Falak! Btw, in the regex above, we can remove ^-?, which means 'may include - sign in the beginning' because we won't take negative numbers, sorry for the confusion.

Reply:
Change implemented. Thanks for the input!